src/schema/v2/me/createCollectionMutation.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  GraphQLObjectType,
+  GraphQLUnionType,
+  GraphQLNonNull,
+  GraphQLString,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { ResolverContext } from "types/graphql"
+import { CollectionType } from "./collection"
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "CreateCollectionSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    collection: {
+      type: CollectionType,
+      resolve: async (response) => {
+        return response
+      },
+    },
+  }),
+})
+
+const ErrorType = new GraphQLObjectType<any, ResolverContext>({
+  name: "CreateCollectionFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => (typeof err.message === "object" ? err.message : err),
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "CreateCollectionResponseOrError",
+  types: [SuccessType, ErrorType],
+})
+
+interface InputProps {
+  name: string
```

Comment:
I've decided to allow only `name` field, even though `/collection` POST endpoint can receive other fields like `description`, `saves`, `private`, and `default`. We need only `name` for our immediate needs, other fields we can add later